router/core/access_controller.go

```diff
@@ -16,15 +33,29 @@ var (
 
 // AccessController handles both authentication and authorization for the Router
 type AccessController struct {
-	authenticationRequired bool
-	authenticators         []authentication.Authenticator
+	authenticationRequired     bool
+	authenticators             []authentication.Authenticator
+	introspectionAuthMode      IntrospectionAuthMode
+	introspectionAuthSkipToken string
 }
 
-func NewAccessController(authenticators []authentication.Authenticator, authenticationRequired bool) *AccessController {
-	return &AccessController{
-		authenticationRequired: authenticationRequired,
-		authenticators:         authenticators,
+// NewAccessController creates a new AccessController.
+// It returns an error if the introspection auth mode is invalid.
+func NewAccessController(
+	authenticators []authentication.Authenticator,
+	authenticationRequired bool,
+	introspectionAuthMode IntrospectionAuthMode,
+	introspectionAuthSkipToken string) (*AccessController, error) {
+	if introspectionAuthMode != IntrospectionAuthModeFull && introspectionAuthMode != IntrospectionAuthModeToken && introspectionAuthMode != IntrospectionAuthModeSkip {
+		return nil, fmt.Errorf("invalid introspection auth mode: %s", introspectionAuthMode)
 	}
+
+	return &AccessController{
+		authenticationRequired:     authenticationRequired,
+		authenticators:             authenticators,
+		introspectionAuthMode:      introspectionAuthMode,
+		introspectionAuthSkipToken: introspectionAuthSkipToken,
+	}, nil
 }
```

Comment:
I think the parameters for this function are now enough that there should likely be an options type, 4 unnamed params of mixed types is a lot.

Reply:
fixed